nncf/quantization/algorithms/bias_correction/algorithm.py

```diff
@@ -182,85 +190,84 @@ def _apply(
-
-        seen_nodes = []
-        nodes_queue = deque(nncf_graph.get_input_nodes())
-        while nodes_queue:
-            current_node = nodes_queue.popleft()
-            current_node_name = current_node.node_name
-
-            if current_node_name in seen_nodes:
-                continue
-
-            seen_nodes.append(current_node_name)
-            if current_node.metatype in self._backend_entity.quantizer_types:
-                target_point = self._backend_entity.target_point(TargetType.LAYER, current_node_name, 0)
-                command = self._backend_entity.node_removing_command(target_point)
-                transformation_layout.register(command)
-            nodes_queue.extend(nncf_graph.get_next_nodes(current_node))
+            # Also, we need to remove unnecessary statistics that we don't need anymore,
+            # to reduce memory usage during the algorithm's pipeline.
+            self._remove_unnecessary_stats(position, subgraphs_data)
 
-        return model_transformer.transform(transformation_layout)
+        return main_model_transformer.transform(main_transformations_layout)
 
-    def _get_subgraph_data_for_node(self, node: NNCFNode, nncf_graph: NNCFGraph) -> Dict[str, List[str]]:
+    def _get_subgraph_data_for_node(self, node: NNCFNode, nncf_graph: NNCFGraph, model: TModel) -> Dict[str, List[str]]:
         """
         This method collects necessary data for the specified node and its subgraph.
         This data contains the nodes (NNCFNode) for the subgraph building
         and statistics collection (for the next step).
 
         :param node: NNCFNode instance. This is the main node that with bias that would be corrected (or not).
         :param nncf_graph: NNCFGraph instance for graph analysis.
+        :param model: Backend-specific model.
         :return: A dict with the list of the nodes for the subgraph input and statistics collection.
         """
-        stats_nodes, input_nodes, output_nodes = [], [], []
-
-        def traverse_to_layers_with_bias(node, output):
-            if node in output:
-                return True, output
-            if self._backend_entity.is_node_with_bias(node, nncf_graph):
-                output.append(node)
-                self._collected_stat_inputs.add(node.node_name)
-                activation_input = nncf_graph.get_input_edges(node)[0].from_node
-
-                output_nodes.append(activation_input)
-                return True, output
-            return False, output
-
-        def traverse_to_input_layers(node, output):
-            if node in output + input_nodes:
-                return True, output
-            if node.node_name in self._collected_stat_inputs and node not in stats_nodes:
-                output.append(node)
-                return True, output
-            return False, output
-
+        statistic_nodes, subgraph_input_nodes, subgraph_output_nodes = [], [], []
+
+        def fill_statistic_nodes(node):
```

Comment:
> Do not nest a function just to hide it from users of a module. Instead, prefix its name with an _ at the module level so that it can still be accessed by tests.

Need move definition of function to class method or separate file.

Reply:
I wouldn't like to implement these functions in the class body since they are not needed there and uses only for recursion inside the method. 
See the `nncf/common/quantization/quantizer_setup.py::equivalent_to`, for example.
Also, since these methods use local variables, we're able to leave it as is.